okio/src/main/kotlin/okio/internal/ByteString.kt

```diff
@@ -168,10 +168,42 @@ internal fun ByteString.commonEndsWith(suffix: ByteString) =
 internal fun ByteString.commonEndsWith(suffix: ByteArray) =
     rangeEquals(size - suffix.size, suffix, 0, suffix.size)
 
+internal fun ByteString.commonIndexOf(other: ByteString, fromIndex: Int): Int {
+  val limit = size - other.size
+  for (i in maxOf(fromIndex, 0)..limit) {
+    if (rangeEquals(i, other, 0, other.size)) {
+      return i
+    }
+  }
+  return -1
+}
+
 internal fun ByteString.commonIndexOf(other: ByteArray, fromIndex: Int): Int {
-  val limit = data.size - other.size
+  val limit = size - other.size
   for (i in maxOf(fromIndex, 0)..limit) {
-    if (arrayRangeEquals(data, i, other, 0, other.size)) {
+    if (rangeEquals(i, other, 0, other.size)) {
+      return i
+    }
+  }
+  return -1
+}
+
+internal fun ByteString.commonLastIndexOf(other: ByteString, fromIndex: Int): Int {
+  var fromIndex = fromIndex
+  fromIndex = minOf(fromIndex, size - other.size)
```

Comment:
you can‘t combine this with the previous line?

Reply:
I copied it from the JVM ByteString implementation, but I'll make it more like the `commonIndexOf` functions.